File: backend/src/main/java/dev/agiro/fanel/recipes/api/RecipesApi.java
package dev.agiro.fanel.recipes.api;

import java.util.List;
import java.util.UUID;

public interface RecipesApi {
    List<RecipeDto> list(UUID householdId);
    RecipeDto get(UUID householdId, UUID recipeId);
    RecipeDto create(UUID householdId, String name, int servings, String notes, String description,
                     List<String> steps, List<String> tags, List<IngredientDto> ingredients,
                     String imageMimeType, String imageData);
    void delete(UUID householdId, UUID recipeId);
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/RecipesApi.kt
package dev.agiro.fanel.android.data.remote

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface RecipesApi {
    @GET("/api/v1/households/{householdId}/recipes")
    suspend fun list(@Path("householdId") householdId: String): List<RecipeDto>

    @GET("/api/v1/households/{householdId}/recipes/{recipeId}")
    suspend fun get(
        @Path("householdId") householdId: String,
        @Path("recipeId") recipeId: String
    ): RecipeDto

    @POST("/api/v1/households/{householdId}/recipes")
    suspend fun create(
        @Path("householdId") householdId: String,
        @Body request: CreateRecipeRequest
    ): RecipeDto

    @DELETE("/api/v1/households/{householdId}/recipes/{recipeId}")
    suspend fun delete(
        @Path("householdId") householdId: String,
        @Path("recipeId") recipeId: String
    )
}
